packs/chef/actions/lib/shell_out.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python2.7
+
+import sys
+import os
+import subprocess
+import select
+
+# Starts command in subprocess processing output as it appears.
+def shell_out(command, env={}):
+    env = os.environ.copy().update(env)
+    proc = subprocess.Popen(args=command, stdin=None, stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE, env=env)
+    while True:
+        fdlist = [proc.stdout.fileno(), proc.stderr.fileno()]
+        ios = select.select(fdlist, [], [])
+        for fd in ios[0]:
+            if fd == proc.stdout.fileno():
+                sys.stdout.write(proc.stdout.read())
```

Comment:
That's a good point - I just quickly glanced over it and I didn't notice it's a remote action. Using st2common won't work with remote actions if st2common is not installed there so yeah, bundling this function inside the pack is fine.